crates/biome_js_analyze/src/semantic_analyzers/nursery/use_export_type.rs

```diff
@@ -0,0 +1,214 @@
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<JsRuleAction> {
+        let export_named_clause = ctx.query();
+        let mut mutation = ctx.root().begin();
+        let type_token =
+            Some(make::token(T![type]).with_trailing_trivia([(TriviaPieceKind::Whitespace, " ")]));
+        let diagnostic = match state {
+            ExportTypeFix::Factorize => {
+                let specifier_list = export_named_clause.specifiers();
+                let mut new_specifiers = Vec::new();
+                for specifier in specifier_list.iter().filter_map(|x| x.ok()) {
+                    if let Some(type_token) = specifier.type_token() {
+                        let mut new_specifier = specifier.with_type_token(None);
+                        if type_token.has_trailing_comments() {
+                            new_specifier = new_specifier.prepend_trivia_pieces(
+                                trim_leading_trivia_pieces(type_token.trailing_trivia().pieces()),
+                            )?;
+                        }
+                        new_specifiers.push(new_specifier);
+                    } else {
+                        new_specifiers.push(specifier)
+                    }
+                }
+                let new_specifier_list = make::js_export_named_specifier_list(
+                    new_specifiers,
+                    specifier_list
+                        .separators()
+                        .filter_map(|sep| sep.ok())
+                        .collect::<Vec<_>>(),
+                );
+                mutation.replace_node(
+                    export_named_clause.clone(),
+                    export_named_clause
+                        .clone()
+                        .with_type_token(type_token)
+                        .with_specifiers(new_specifier_list),
+                );
+                JsRuleAction {
+                    category: ActionCategory::QuickFix,
+                    applicability: Applicability::Always,
+                    message: markup! { "Use a grouped "<Emphasis>"export type"</Emphasis>"." }
+                        .to_owned(),
+                    mutation,
+                }
+            }
+            ExportTypeFix::AddInlineType(specifiers) => {
+                for specifier in specifiers {
+                    mutation.replace_node(
+                        specifier.clone(),
+                        specifier.clone().with_type_token(type_token.clone()),
+                    );
+                }
+                JsRuleAction {
+                    category: ActionCategory::QuickFix,
+                    applicability: Applicability::Always,
+                    message: markup! { "Use inline "<Emphasis>"type"</Emphasis>" exports." }
+                        .to_owned(),
+                    mutation,
+                }
+            }
+        };
+        Some(diagnostic)
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum ExportTypeFix {
+    Factorize,
```

Comment:
I am not sure this word can be applied in this context. It's worth at least giving us some documentation to explain what this variant stands for